test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -2482,4 +2483,61 @@ public function testGetAttributesByInstance(): void
 
         self::assertCount(2, $attributes);
     }
+
+    public function testBugWithTraitMethodVisibilityOverride(): void
+    {
+        $php            = <<<'PHP'
+            <?php
+
+            interface Foo
+            {
+                public function doFoo(): void;
+            }
+
+            class Bar implements Foo
+            {
+                public function doFoo(): void
+                {
+                    echo "Doing foo the generic way";
```

Comment:
Is it possible to reduce this code even further? I suppose method body isn't that relevant, right?

Reply:
Yeah, it's irrelevant, I just copied it from the PHPStan bug report :) Gonna amend that.

Reply:
Figured: just want to have the test as-on-point-as-possible, before merging 👍

Reply:
Done, verified it still fails without the `clone`.